src/utils/validateBurnedAmount/validateAvaxBurnedAmountEtna.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  isAddPermissionlessDelegatorTx,
+  isAddPermissionlessValidatorTx,
+  isAddSubnetValidatorTx,
+  isCreateChainTx,
+  isCreateSubnetTx,
+  isPvmBaseTx,
+  isExportTx as isPvmExportTx,
+  isImportTx as isPvmImportTx,
+  isRemoveSubnetValidatorTx,
+  isTransferSubnetOwnershipTx,
+} from '../../serializable/pvm';
+import type { UnsignedTx } from '../../vms/common';
+
+/**
+ * Validate burned amount for post-etna avalanche transactions
+ *
+ * @param unsignedTx: unsigned transaction
+ * @param burnedAmount: burned amount in nAVAX
+ * @param baseFee: pvm dynamic fee caculator, https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/txs/fee/dynamic_calculator.go
+ * @param feeTolerance: tolerance percentage range where the burned amount is considered valid. e.g.: with FeeTolerance = 20% -> (expectedFee <= burnedAmount <= expectedFee * 1.2)
+ * @return {boolean} isValid: : true if the burned amount is valid, false otherwise.
+ * @return {bigint} txFee: burned amount in nAVAX
+ */
+export const validateAvaxBurnedAmountEtna = ({
+  unsignedTx,
+  burnedAmount,
+  baseFee,
+  feeTolerance,
+}: {
+  unsignedTx: UnsignedTx;
+  burnedAmount: bigint;
+  baseFee: bigint;
+  feeTolerance: number;
+}): { isValid: boolean; txFee: bigint } => {
+  const tx = unsignedTx.getTx();
+
+  const feeToleranceInt = Math.floor(feeTolerance);
+
+  if (feeToleranceInt < 1 || feeToleranceInt > 100) {
+    throw new Error('feeTolerance must be [1,100]');
+  }
+
+  const min = baseFee;
```

Comment:
I don't think this logic is correct. We will pass in the gas price for baseFee, right? You should likely pull in a reference to the `src/vms/pvm/txs/fee/calculator.ts` calculator. We should follow the logic more closely to the evm handler, which is doing something very similar but using a different cost calculation function. In fact, you might just re-use the same exact method for both -- the one additional parameter would be the cost calculation function? Not sure, just you might not need separate functions for this.

Reply:
merged the logic into validateDynamicBurnedAmont.ts